src/llamacrawl/services/sessions/source_router.py

```
@@ -0,0 +1,87 @@
+        Returns:
+            Tuple of (source_type, reader_class)
+
+        Raises:
+            ValueError: If path doesn't match any configured source
+        """
+        if config.claude_code_path and path.is_relative_to(config.claude_code_path.expanduser()):
+            return ("claude_code", ClaudeCodeReader)
+        elif config.codex_path and path.is_relative_to(config.codex_path.expanduser()):
+            return ("codex", CodexReader)
+        else:
+            raise ValueError(f"File path {path} does not match any configured session source")
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)